router/router.go

```diff
@@ -0,0 +1,229 @@
+package router
+
+import (
+	"encoding/json"
+	"errors"
+	"iotdashboard/controller"
+	"log"
+	"net"
+	"net/http"
+)
+
+type RouterService struct {
+	Ctrlr     controller.ControllerService
+	httpPort  string
+	httpsPort string
+	certPath  string
+	keyPath   string
+}
+
+// Credentials is a struct that holds the email, password, and CSRF token of a request
+type Credentials struct {
+	Email    string `json:"email"`
+	Password string `json:"password"`
+	CSRF     string `json:"csrf"`
+}
+
+func NewRouter(httpPort, httpsPort, certPath, keyPath string) (RouterService, error) {
+	Ctrlr, err := controller.NewController()
+	if err != nil {
+		return RouterService{}, err
+	}
+	return RouterService{Ctrlr, httpPort, httpsPort, certPath, keyPath}, nil
+}
+
+func (rtr *RouterService) Start() error {
+	log.Printf("Starting webserver ... \n")
+	//start listening for http to redirect to https
+	go log.Print(http.ListenAndServe(rtr.httpPort, http.HandlerFunc(rtr.redirectTLS)))
+
+	//start listening for https and handle requests
+	err := rtr.handleRequests(rtr.certPath, rtr.keyPath)
+	if err != nil {
+		log.Printf("HandleRequests Error: %v \n", err)
+		return err
+	}
+	return nil
+}
+
+func (rtr *RouterService) handleRequests(certPath, keyPath string) error {
+	mux := http.NewServeMux()
+	mux.Handle("/", http.FileServer(http.Dir("iotdbfrontend/build/")))
+	mux.HandleFunc("/login", rtr.loginHandler)
+	mux.HandleFunc("/logout", rtr.logoutHandler)
+	mux.HandleFunc("/csrf", rtr.csrfHandler)
+
+	log.Printf("Running! \n")
+	err := http.ListenAndServeTLS(rtr.httpsPort, certPath, keyPath, mux)
+	if err != nil {
+		log.Printf("ListenAndServeTLS Error: %v /n", err)
+		return err
+	}
+	return nil
+}
+
+func (rtr *RouterService) loginHandler(w http.ResponseWriter, r *http.Request) {
+	log.Printf("Received POST/login request")
+	rtr.addHeaders(w)
+	if r.Method != "POST" {
+		http.Error(w, "Method not supported", http.StatusMethodNotAllowed)
+		return
+	}
+	var creds Credentials
+	// Get the JSON body and decode into credentials
+	err := json.NewDecoder(r.Body).Decode(&creds)
+	if err != nil {
+		log.Printf("LoginHandler Error: %v /n", err)
+		http.Error(w, "Bad Request", http.StatusBadRequest)
+		return
+	}
+
+	// CSRF Validation
+	// CSRF Validation
+	err = rtr.validateCSRF(w, r, creds)
+	if err != nil {
+		log.Printf("CSRF Validation err: %v", err)
+		return
+	}
+	log.Printf("CSRF Validated")
+
+	//Perform Login
+	jwt, err := rtr.Ctrlr.Login(creds.Email, creds.Password)
+	if err != nil {
+		http.Error(w, "Email and Password do not match", http.StatusUnauthorized)
+		return
+	}
+	log.Printf("Logged In")
+
+	http.SetCookie(w, &http.Cookie{
+		Name:     "JWT",
+		Value:    jwt,
+		Secure:   true,
+		HttpOnly: true,
+		SameSite: http.SameSiteStrictMode,
+	})
+	log.Printf("Responding")
```

Comment:
These logs feel like debug leftovers. They don't add much value to the logs without additional context so I would consider to either remove them or add more context so they're useful.